packages/edit-site/src/components/styles-canvas/index.js

```diff
@@ -0,0 +1,156 @@
+import { useFocusOnMount, useFocusReturn } from '@wordpress/compose';
+import { store as preferencesStore } from '@wordpress/preferences';
+import {
+	store as editorStore,
+	privateApis as editorPrivateApis,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { store as editSiteStore } from '../../store';
+import { StylesCanvasStyleBook } from './style-book';
+import { StylesCanvasRevisions } from './revisions';
+
+const { EditorContentSlotFill, ResizableEditor } = unlock( editorPrivateApis );
+
+/**
+ * Helper function to get the title for the styles canvas based on current state.
+ *
+ * @param {string}  path          Current styles path.
+ * @param {boolean} showStylebook Whether stylebook is visible.
+ * @return {string} Translated string for the canvas title.
+ */
+export function getStylesCanvasTitle( path, showStylebook ) {
+	if ( showStylebook ) {
+		return __( 'Style Book' );
+	}
+	if ( path?.startsWith( '/revisions' ) ) {
+		return __( 'Style Revisions' );
+	}
+	return '';
+}
+
+/**
+ * Styles canvas component - orchestrates rendering of style book and revisions.
+ * Determines what content to show based on global styles navigation state.
+ *
+ * @return {JSX.Element|null} The styles canvas or null if nothing to render.
+ */
+export default function StylesCanvas() {
+	const { stylesPath, showStylebook, showListViewByDefault } = useSelect(
+		( select ) => {
+			const { getStylesPath, getShowStylebook } = unlock(
+				select( editSiteStore )
+			);
+
+			const _showListViewByDefault = select( preferencesStore ).get(
+				'core',
+				'showListViewByDefault'
+			);
+
+			return {
+				stylesPath: getStylesPath(),
+				showStylebook: getShowStylebook(),
+				showListViewByDefault: _showListViewByDefault,
+			};
+		},
+		[]
+	);
+	const { resetStylesNavigation, setStylesPath } = unlock(
+		useDispatch( editSiteStore )
+	);
+	const { setIsListViewOpened } = useDispatch( editorStore );
+
+	const focusOnMountRef = useFocusOnMount( 'firstElement' );
+	const sectionFocusReturnRef = useFocusReturn();
+
+	function onCloseCanvas() {
+		setIsListViewOpened( showListViewByDefault );
+		resetStylesNavigation();
+	}
+
+	function closeOnEscape( event ) {
+		if ( event.keyCode === ESCAPE && ! event.defaultPrevented ) {
+			event.preventDefault();
+			onCloseCanvas();
+		}
+	}
+
+	// Determine what content to render
+	let content = null;
+
+	if ( showStylebook ) {
+		content = (
+			<StylesCanvasStyleBook
+				path={ stylesPath }
+				onPathChange={ setStylesPath }
+			/>
+		);
+	} else if ( stylesPath?.startsWith( '/revisions' ) ) {
+		content = <StylesCanvasRevisions path={ stylesPath } />;
+	}
+
+	const title = getStylesCanvasTitle( stylesPath, showStylebook );
+
+	if ( ! content ) {
+		return null;
+	}
+
+	// Clone content with ref to allow proper height inheritance
+	const contentWithRef = cloneElement( content, {
+		ref: sectionFocusReturnRef,
+	} );
```

Comment:
I think we need to forward refs right down to the stylebook/revisions otherwise the console is spammed with `Function components cannot be given refs` errors